includes/class-wc-session-handler.php

```diff
@@ -63,6 +66,7 @@ public function __construct() {
 		}
 
 		$this->_data = $this->get_session_data();
+		$this->_data_org = serialize($this->_data);
```

Comment:
I've changed the serialize calls to json_encode and tested it.
This seems to work fine.

The data to json_encode is already pre-serialized, encoding it returns something like the following json string:
```
{
	"cart" : "a:1:{s:32:\"48df7b8e8d586a55cf3e7054a4c85b30\";a:9:{s:10:\"product_id\";i:3752;s:12:\"variation_id\";i:0;s:9:\"variation\";a:0:{}s:8:\"quantity\";i:11;s:10:\"line_total\";d:23.100000000000001;s:8:\"line_tax\";d:1.3859999999999999;s:13:\"line_subtotal\";d:23.100000000000001;s:17:\"line_subtotal_tax\";d:1.3859999999999999;s:13:\"line_tax_data\";a:2:{s:5:\"total\";a:1:{i:1;d:1.3859999999999999;}s:8:\"subtotal\";a:1:{i:1;d:1.3859999999999999;}}}}",
	"applied_coupons" : "a:0:{}",
	"coupon_discount_amounts" : "a:0:{}",
	"coupon_discount_tax_amounts" : "a:0:{}",
	"removed_cart_contents" : "a:1:{s:32:\"48df7b8e8d586a55cf3e7054a4c85b30\";a:9:{s:10:\"product_id\";i:3752;s:12:\"variation_id\";i:0;s:9:\"variation\";a:0:{}s:8:\"quantity\";i:3;s:10:\"line_total\";d:6.3000000000000007;s:8:\"line_tax\";d:0.378;s:13:\"line_subtotal\";d:6.3000000000000007;s:17:\"line_subtotal_tax\";d:0.378;s:13:\"line_tax_data\";a:2:{s:5:\"total\";a:1:{i:1;d:0.378;}s:8:\"subtotal\";a:1:{i:1;d:0.378;}}}}",
	"cart_contents_total" : 23.1,
	"total" : 0,
...
}
```

So we can see the array to serialize consists of already serialized arrays or numbers.
This is good news, it doesn't seem to matter if we encode using json or the PHP serialize format.